Add Tips section for AOT enabled builds (#40770)
dotnet/docs: docs/core/extensions/options-validation-generator.md

:::code source="snippets/configuration/console-validation-gen/Program.cs":::

> [!TIP]

> > When AOT compilation is enabled by including `<PublishAot>true</PublishAot>` in the csproj file, the code may generate warnings such as [IL2025](../deploying/trimming/trim-warnings/il2025.md) and [IL3050](../deploying/native-aot/warnings/il3050.md). It's recommended to utilize the configuration source generator to mitigate these warnings. To enable the configuration source generator, add the property `<EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>` to the project file.

Reviewer: ```suggestion
> When AOT compilation is enabled by including `<PublishAot>true</PublishAot>` in the _.csproj_ file, the code may generate warnings such as [IL2025](../deploying/trimming/trim-warnings/il2025.md) and [IL3050](../deploying/native-aot/warnings/il3050.md). It's recommended to utilize the configuration source generator to mitigate these warnings. To enable the configuration source generator, add the property `<EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>` to the project file.
```

> > When AOT compilation is enabled by including `<PublishAot>true</PublishAot>` in the _.csproj_ file, the code may generate warnings such as [IL2025](../deploying/trimming/trim-warnings/il2025.md) and [IL3050](../deploying/native-aot/warnings/il3050.md). It's recommended to utilize the configuration source generator to mitigate these warnings. To enable the configuration source generator, add the property `<EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>` to the project file.

Reviewer 2: ```suggestion
> When AOT compilation is enabled by including `<PublishAot>true</PublishAot>` in the _.csproj_ file, the code might generate warnings such as [IL2025](../deploying/trimming/trim-warnings/il2025.md) and [IL3050](../deploying/native-aot/warnings/il3050.md). To mitigate these warnings, it's recommended to use the configuration source generator. To enable the configuration source generator, add the property `<EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>` to the project file.
```
